### DISCUSSION TITLE
Unable to subscribe mempool via websocket

### DISCUSSION THREAD

**Discussion by anggapur** [0xPolygon/polygon-edge#868] [Q&A]
Hello, I'm currently working on a project that create a webhook, and this webhook will be able to listen to pending transactions (mempool). But I'm unable to find any clue about subscribe mempool on polygon edge. 

I try to subscribe polygon edge's mempool with ethers js, here my code
```
const WS_URL = '{My Polygon Websocket Here}';
const provider =  new ethers.providers.WebSocketProvider(WS_URL);

provider.on("pending", (tx) => {
    // Emitted when any new pending transaction is noticed
});
```

Note: this code works on public chain (tested on mumbai & goerli)

But it return errors when I try to subscribe to polygon edge, here the error that I got :

```
Error : subscribe method newPendingTransactions not found
```

After a short research, I found that it seems like polygon edge not yet support subscription for pending transactions  ( newPendingTransactions method not supported yet )
https://github.com/0xPolygon/polygon-edge/blob/cac8de6f8e48e09d2cb748dea24de4af18b500d7/jsonrpc/dispatcher.go#L165-L175

is there any other way or trick to listen to polygon edge mempool?

**Comment by laviniat1996** [comment]
Hi @anggapur ,

We support Tx Pool subscription through gRPC. You can find a list of flags for subcription [here](https://wiki.polygon.technology/docs/edge/get-started/cli-commands/#transaction-pool-commands), in the **txpool subscribe flags** category.

**Reply by anggapur** [reply]
Hi @laviniat1996 Thank you for the answer.
But it seems better if we had Tx pool subscription on Websocket like other node provider on public chain

**Reply by anggapur** [reply]
Hi @laviniat1996 Thank you for the answer.
But it seems better if we had Tx pool subscription on Websocket like other node provider on public chain

**Reply by rachit77** [reply]
Hi @anggapur,
I was curious, Why is a Tx pool subscription on WebSocket better than Http2(gRPC)?
